parsel/csstranslator.py

```diff
@@ -38,46 +48,73 @@ def __str__(self):
 
         return path
 
-    def join(self, combiner, other, *args, **kwargs):
+    def join(
+        self: _XPathExprType,
+        combiner: str,
+        other: OriginalXPathExpr,
+        *args: Any,
+        **kwargs: Any,
+    ) -> _XPathExprType:
+        assert isinstance(other, XPathExpr)
```

Comment:
I would only use asserts in scenarios where removing the assert would be OK. Here raising a `ValueError` may be better.